### ISSUE TITLE
Do a final log check for related actions before protection and block actions

### ISSUE THREAD

**Issue by danielquinlan** [wikimedia-gadgets/twinkle#1940]
I realize there's an unavoidable race condition (at least without API changes) with many administrator actions, but I see a lot of overlapping actions where the time between two administrators making the same or similar actions is more than a few seconds, but less than about 5 minutes (usually it's just a minute or two because that's how long it sometimes takes to customize a complex protection reason or block reason in the Twinkle dialog). The time is not really long enough for a human to avoid unless you're continually refreshing multiple pages, but it's very easily detected by Twinkle.

What I'm requesting is a quick final check right when someone pushes the final protect/unprotect button or the block button that checks the log and if a **different** administrator has an related action in the last 5 minutes (maybe even 10 minutes). If there is such an action put up a dialog showing the action and allow the administrator to cancel.

**Comment by NovemLinguae** [comment]
> if a different administrator has an related action in the last 5 minutes

Just to clarify, I assume you mean "if a different administrator performs the same action (block, protect) to the same user/page in the last 5 minutes"?

**Comment by danielquinlan** [comment]
Yes. I would say the same action or the inverse action. So, for example, if the user is protecting a page and there's an recent unprotection action, pop up the dialog for that too. The same for blocks/unblocks.
